app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/summary/SummaryProductFragment.java

```diff
@@ -368,6 +373,16 @@ public void refreshView(State state) {
             mUrlImage = product.getImageUrl();
         }
 
+        yourListedProductsDao = Utils.getAppDaoSession(getContext()).getYourListedProductDao();
+        //check if this product exists in eaten list
+        if (checkIfProductIsEaten()) {
+            btnEating.setVisibility(View.GONE);
```

Comment:
please, is it possible to add a new method
updateEatingButtons(boolean isEaten){
...
} 
and replace the lines 378-384 by updateEatingButton(isProductEatenByCurrentUser())
and then 1053-1054 updateEatingButtons(true)
and then 1060-1061 updateEatingButtons(false)